File: EarthquakeList/src/main/java/com/jsbomb/earthquakelist/actors/GeoJsonObjectParser.java
package com.jsbomb.earthquakelist.actors;

import com.google.gson.Gson;
import com.jsbomb.earthquakelist.data.EarthquakesSummary;

/**
 * Created by john.seo on 5/17/2017.
 */

public class GeoJsonObjectParser extends DataParser {

    @Override
    public void parse() {
        boolean success = true;
        String jsonString = (String)getData();
        EarthquakesSummary earthquakesSummary = null;

        try {
            earthquakesSummary = new Gson().fromJson(jsonString, EarthquakesSummary.class);
        } catch (Exception ex) {
            ex.printStackTrace();
            success = false;
        }

        OnParseCompleteListener listener = getOnParseCompleteListener();
        if (listener != null) {
            listener.onParseComplete(success, earthquakesSummary);
        }
    }
}

File: EarthquakeList/src/main/java/com/jsbomb/earthquakelist/actors/GeoJsonObjectParser.kt
package com.jsbomb.earthquakelist.actors

import com.google.gson.Gson
import com.jsbomb.earthquakelist.data.EarthquakesSummary

class GeoJsonObjectParser : DataParser() {

    override fun parse() {
        var success = true
        val jsonString = getData() as String
        var earthquakesSummary: EarthquakesSummary? = null

        try {
            earthquakesSummary = Gson().fromJson(jsonString, EarthquakesSummary::class.java)
        } catch (ex: Exception) {
            ex.printStackTrace()
            success = false
        }

        getOnParseCompleteListener()?.onParseComplete(success, earthquakesSummary)
    }
}